### PR TITLE
feat: add in error handling and resolutions for empty TCP pipes

### PR DESCRIPTION
**Pull request by Serpent03**
Add in error handling and `println!()` logs for clarity(for the time being).

Fix hangups from empty TCP pipes(not tested fully).

### CODE REVIEW

File: packages/tunlrs-core/src/tunnel/tunnelconnection.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-use ::core::net::SocketAddr;
+use ::std::net::SocketAddr;
```
Comment (Serpent03) [inline packages/tunlrs-core/src/tunnel/tunnelconnection.rs:1]: shift from ``::core::net` to `::std::net`. As far as I understand, the former is for nightly builds and we will have to opt in for that through a macro.

Not sure if that would be desirable, so I switched to using `std::net`.

**Comment by sohamjaiswal** [comment]
No, dont use println, shift to a proper logging package, m open to using any crate like tracing, slog or log

**Comment by Serpent03** [comment]
> No, dont use println, shift to a proper logging package, m open to using any crate like tracing, slog or log

Then tracing is probably a better approach to logging(atleast intermediately) since it supports concurrent operation logging

Should we open another issue and then submit a PR in relation to that for enabling logging globally?

**Comment by sosweetham** [comment]
> > No, dont use println, shift to a proper logging package, m open to using any crate like tracing, slog or log
> 
> Then tracing is probably a better approach to logging(atleast intermediately) since it supports concurrent operation logging
> 
> Should we open another issue and then submit a PR in relation to that for enabling logging globally?

sounds good to me

**Comment by sosweetham** [comment]
just open an issue and i'll just link it here for reference, after that i'll merge

**Comment by Serpent03** [comment]
Merged and deleted after opening issue #18

**Pull request merged**